R/filterFeatures.R

```diff
@@ -42,7 +46,21 @@ filterFeatures = function(task, method = "randomForestSRC.rfsrc", fval = NULL, p
   )
 
   if (is.null(fval)) {
-    fval = generateFilterValuesData(task = task, method = method, nselect = nselect, ...)$data
+    if (isTRUE(cache)) {
+
+      if (!requireNamespace("memoise", quietly = TRUE)) {
+        stop("Package \"memoise\" needed for this function to work. Please install it.",
+             call. = FALSE)
+      }
+
+      # caching calls to `generateFilterValuesData()` with the same arguments
+      cache = memoise::cache_filesystem("~/.cache")
```

Comment:
Does this work on Windows? What if the directory doesn't exist? I would make this an option, i.e. instead of a Boolean make `cache` a character with `NULL` meaning no caching, otherwise use that directory. And if it's empty, use the system temp dir `tempdir()`.

Reply:
Also if the cache was saved as part of the learner (instead of just the Boolean) repeated uses of the learner would benefit from the caching.

Reply:
> Does this work on Windows? What if the directory doesn't exist?

Good point. Didn't check on this. Will do it now.

>  instead of a Boolean make cache a character with NULL meaning no caching, otherwise use that directory.

Sounds good. 

> And if it's empty, use the system temp dir tempdir().

Tried tempdir() first and faced some weird errors, that's why I went with `~/.cache`. 

> Also if the cache was saved as part of the learner (instead of just the Boolean) repeated uses of the learner would benefit from the caching.

There are more opportunities for which we could use caching. But I think we should do this in a new PR.